Develop

VagapovaLi opened the pull request.

VagapovaLi committed:
добавление в  develop

VagapovaLi committed:
add allure report

VagapovaLi committed:
Merge remote-tracking branch 'origin/main' into develop

VagapovaLi committed:
add locators, pages and tests

klushcheva commented on pages/home_page.py:
```diff
@@ -0,0 +1,47 @@
+
+
+class Dropdownlist:
+
```
Нужно поправить: не реализована РОМ. Все классы страниц наследуются от BasePage; объекты BasePage в тестах не создаются, только объекты классов-наследников

klushcheva commented on test/test_home_page.py:
```diff
@@ -0,0 +1,89 @@
+        self.driver.quit()
+
+    @allure.step('Прокрутка к вопросам, окрыть первый вопрос, ответ')
+    def test_check_click_1_question(self, driver):
```
Нужно поправить: вместо 8 отдельных тестов нужно написать один параметризованный тест: логика идентична, отличаются только локаторы и тестовые данные

klushcheva commented on test/test_order_page.py:
```diff
@@ -0,0 +1,57 @@
+
+        order_page.order(name=name, surname=surname, address=address, phone_number=phone_number, comment=comment)
+
+        success_message = driver.find_element(*OrderPageLocators.TEXT_ORDER_SUBMITTED).text
```
Нужно поправить: прямых вызовов драйвера не должно быть, всю логику заворачиваем в методы пейджей

klushcheva commented on test/test_order_page.py:
```diff
@@ -0,0 +1,57 @@
+
+        order_page.check_click_view_status()
+        base_page.click_on_the_yandex_logo_and_switch_window()
+        assert self.driver.current_url == DZEN_URL, f"Ожидался URL: {DZEN_URL}, но получен: {self.driver.current_url}"
```
Тут аналогично с вызовом драйвера

VagapovaLi committed:
Исправлена реализация POM
Исправлено наследование от BasePage
Написан параметризованый тест вместо 8 отдельных тетсов
Убрала вызов драйвера из тестов.
